web/src/pages/indicators/[slug].page.tsx

```diff
@@ -0,0 +1,221 @@
+				.split("/")
+				.pop()
+				?.toLowerCase()}`,
+		});
+	});
+
+	return chaptersAndLinksArray;
+};
+
+export type IndicatorsDetailsPageProps = {
+	slug: string;
+	product: ProductDetail;
+	productType: ProductType;
+};
+
+export default function IndicatorsDetailsPage({
+	product,
+	productType,
+	slug,
+}: IndicatorsDetailsPageProps): JSX.Element {
+	let chapters;
+
+	if (product.chapterHeadings) {
+		chapters = chaptersAndLinks(product.summary, product.chapterHeadings, slug);
+	}
+
+	const nextPageLink = chapters?.[1];
+
+	const metaData = [
+		productType.name,
+		product.id,
+		product.publishedDate ? (
+			<>
+				Published:
+				<time dateTime={dayjs(product.publishedDate).format("YYYY-MM-DD")}>
+					&nbsp;{formatDateStr(product.publishedDate)}
+				</time>
+			</>
+		) : null,
+		product.lastMajorModificationDate != product.publishedDate ? (
+			<>
+				Last updated:
+				<time dateTime={dayjs(product.lastModified).format("YYYY-MM-DD")}>
+					&nbsp;{formatDateStr(product.lastModified)}
+				</time>
+			</>
+		) : null,
+	].filter(Boolean);
+
+	return (
+		<>
+			<NextSeo
+				title={
+					product.title + " | Indicators | Standards and Indicators | NICE"
+				}
+				description={product.metaDescription}
+				additionalLinkTags={[
+					{
+						rel: "sitemap",
+						type: "application/xml",
+						href: "/indicators/sitemap.xml",
+					},
+				]}
+			/>
+
+			<Breadcrumbs>
+				<Breadcrumb to="/">Home</Breadcrumb>
+				<Breadcrumb to="/standards-and-indicators">
+					Standards and Indicators
+				</Breadcrumb>
+				<Breadcrumb to="/standards-and-indicators/indicators">
+					Indicators
+				</Breadcrumb>
+				<Breadcrumb>{product.id}</Breadcrumb>
+			</Breadcrumbs>
+
+			<PageHeader
+				heading={product.title}
+				useAltHeading
+				id="content-start"
+				metadata={metaData}
+			/>
+
+			<Grid gutter="loose">
+				{chapters ? (
+					<GridItem
+						cols={12}
+						md={4}
+						lg={3}
+						elementType="section"
+						aria-label="Chapters"
+					>
+						<PublicationsChapterMenu chapters={chapters} />
+					</GridItem>
+				) : null}
+				<GridItem cols={12} md={8} lg={9} elementType="section">
+					{product.summary ? (
+						<div
+							dangerouslySetInnerHTML={{ __html: product.summary }}
+							className={styles.summary}
+						/>
+					) : null}
+					{nextPageLink ? (
+						<PrevNext
+							nextPageLink={{
+								text: nextPageLink.title,
+								destination: nextPageLink.url,
+								elementType: ({ children, ...props }) => (
+									<Link {...props} scroll={false}>
+										{children}
+									</Link>
+								),
+							}}
+						/>
+					) : null}
+				</GridItem>
+			</Grid>
+		</>
+	);
+}
+
+export const getServerSideProps: GetServerSideProps<
+	IndicatorsDetailsPageProps
+> = async ({ params }) => {
+	if (
+		!params ||
+		!params.slug ||
+		Array.isArray(params.slug) ||
+		!params.slug.includes("-")
+	) {
+		return { notFound: true };
+	}
+
+	const productTypes = await getAllProductTypes();
```

Comment:
I think it would be more efficient to fire the requests (product types and product details) off in parallel, then `await` them both rather than sequentially.

Reply:
I've refactored this, need to check it's actually working in parallel :-)